packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -532,18 +537,21 @@ export class PipelineEditor extends React.Component<
     );
 
     if (id === 'browse_file') {
+      const currentExt = path.extname(filename);
       showBrowseFileDialog(this.browserFactory.defaultBrowser.model.manager, {
         startPath: path.dirname(filename),
         filter: (model: any): boolean => {
-          return model.type == 'notebook';
+          const ext = path.extname(model.path);
+          return currentExt === ext;
```

Comment:
Yes. Users won't be able change an `ipynb` node to a `python` node and vice versa. Several reasons, but the main one (why I lobbied for locking this down) is that at some point in the future a node for file type X might have properties that don't apply to file type Y. 

The only disadvantage I could think of is that with this approach one can't switch a node from one type to another without having to re-provide the "shared" properties (which as of 1.2 are all properties exposed in the UI). However, since switching node types from one to the other is likely going to happen rather infrequently I don't believe this a major inconvenience.

Reply:
yes, I believe really soon the property sets are going to be different for different node types, runtimes, etc... so validation and morphing can/will become a nightmare.